Update feature stability for 1.2 (#4434)
istio/istio.io: content/about/feature-stages/index.md

| Resilience features: timeouts, retries, connection pools, outlier detection | Stable
| Gateway: Ingress, Egress for all protocols | Stable
| TLS termination and SNI Support in Gateways | Stable
| SNI (multiple certs) at ingress | Stable
| [Locality load balancing](/docs/ops/traffic-management/locality-load-balancing/) | Beta
| Enabling custom filters in Envoy | Alpha
| CNI container interface | Alpha
| [Sidecar API](/docs/reference/config/networking/v1alpha3/sidecar/) | Alpha

> The `Sidecar` runtime is considered Beta, but it is advertised as Alpha since its API is still subject to backward incompatible change.

Reviewer: ```suggestion
The `Sidecar` runtime is considered Beta. However, its API is still subject to a backwards incompatible change.
Due to this, we advertise it as Alpha.
```

Author: done!

| [Client and Server Telemetry Reporting](/docs/concepts/policies-and-telemetry/) | Stable
| [Service Dashboard in Grafana](/docs/tasks/telemetry/metrics/using-istio-dashboard/) | Stable
| [Istio Component Dashboard in Grafana](/docs/tasks/telemetry/metrics/using-istio-dashboard/) | Stable

> | [Distributed Tracing to Zipkin / Jaeger](/docs/tasks/telemetry/distributed-tracing/) | Stable

Reviewer 2: Zipkin causes a lot of confusion in the telemetry issues.  I think this would be better worded as `Distributed tracing using the ZipKin API to Jaeger`

thoughts @douglas-reid ?

Reviewer 3: The difference between this line and the one below it are hard to parse out. I'd suggest collapsing and just calling it Distributed Tracing.

Author: Changed to just Distributed Tracing

| [Service Tracing](/docs/tasks/telemetry/distributed-tracing/) | Stable
| [Stackdriver Integration](/docs/reference/config/policy-and-telemetry/adapters/stackdriver/) | Beta
| [Logging with Fluentd](/docs/tasks/telemetry/logs/fluentd/) | Beta
| [Trace Sampling](/docs/tasks/telemetry/distributed-tracing/overview/#trace-sampling) | Beta

> | Prometheus Operator | Alpha

Reviewer 3: I'd maybe call this `Prometheus Operator Support`.

Reviewer 4: is this new?  I don't recall a doc page for it, neither in my k8s cluster when istio default profile is installed @douglas-reid

Reviewer 4: @howardjohn reminded me this is for new installer, so i think we should not mention this for 1.2

Author: Removed mention of the Prometheus Operator

| [Kubernetes: Service Credential Distribution](/docs/concepts/security/#pki)   | Stable
| [VM: Service Credential Distribution](/docs/concepts/security/#pki)         | Beta
| [Mutual TLS Migration](/docs/tasks/security/mtls-migration)    | Beta
| [Cert management on Ingress Gateway](/docs/tasks/traffic-management/ingress/secure-ingress-sds) | Beta
| [Authorization (RBAC)](/docs/concepts/security/#authorization)   | Alpha
| [End User (JWT) Authentication](/docs/concepts/security/#authentication)  | Alpha
| [OPA Checker](/docs/reference/config/policy-and-telemetry/adapters/opa/)    | Alpha
| [Vault integration](/docs/tasks/security/vault-ca) | Alpha
| [SDS Integration](/docs/tasks/security/auth-sds/) | Alpha

> The 'Authorization (RBAC)' runtime is considered Beta, but it is advertised as Alpha since its API is still subject to backward incompatible change.

Author: @liminw @linsun I'm proposing this as a way of indicating that the impl is more stable, but the API is still subject to change

Reviewer 5: Sounds good to me. Thanks @duderino

Reviewer: @duderino Maybe this wording would work better:
```suggestion
The 'Authorization (RBAC)' runtime is considered Beta. However, its API is still subject to a backwards incompatible change.
Due to this, we advertise it as Alpha.
```

| [Kubernetes: Service Credential Distribution](/docs/concepts/security/#pki)   | Stable
| [VM: Service Credential Distribution](/docs/concepts/security/#pki)         | Beta
| [Mutual TLS Migration](/docs/tasks/security/mtls-migration)    | Beta

> | [Authorization (RBAC)](/docs/concepts/security/#authorization)   | Beta

Reviewer 4: do we want to call this beta?  I feel the new API is under reviewed last week.

Author: Set it back to Alpha and added a note that the impl could be considered Beta but the API is still subject to change like an Alpha

| [Kubernetes: Envoy Installation and Traffic Interception](/docs/setup/kubernetes/)        | Stable
| [Kubernetes: Istio Control Plane Installation](/docs/setup/kubernetes/) | Stable

> | [Attribute Expression Language](/docs/reference/config/policy-and-telemetry/expression-language/)        | Stable

Reviewer 2: instead of saying Deprecated, would it make sense to state `Deprecated and removed` so operators know a deprecated feature is no longer available?  In 1.3, we could move the line entirely.

Author: Happy to comply.  @douglas-reid thoughts?

Author: Happy to comply, @douglas-reid thoughts?

Reviewer 3: I don't think we need to keep it. It was `Deprecated` one release ago. No new in-process adapters have been accepted. And we've been actively rejecting them for (most of) 2 release cycles. I think it is time to just remove it.

Reviewer 4: agreed, we should just remove it

Author: removed
